daemon/k8s_watcher.go

```
@@ -437,3 +471,171 @@ func (d *Daemon) syncLB(newSN, modSN, delSN *types.K8sServiceNamespace) {
 		addSN(*newSN)
 	}
 }
+
+func (d *Daemon) ingressAddFn(obj interface{}) {
+	ingress, ok := obj.(*v1beta1.Ingress)
+	if !ok {
+		return
+	}
+
+	if ingress.Spec.Backend == nil {
+		// We only support Single Service Ingress for now
+		return
+	}
+
+	svcName := types.K8sServiceNamespace{
+		Service:   ingress.Spec.Backend.ServiceName,
+		Namespace: ingress.Namespace,
+	}
+
+	d.loadBalancer.K8sMU.Lock()
+	defer d.loadBalancer.K8sMU.Unlock()
+
+	ingressPort := ingress.Spec.Backend.ServicePort.IntValue()
+	fePort, err := types.NewFEPort(types.TCP, uint16(ingressPort))
+	if err != nil {
+		return
+	}
+
+	var host net.IP
+	if d.conf.IPv4Disabled {
+		host = d.conf.HostV6Addr
+	} else {
+		host = d.conf.HostV4Addr
```

This would assume that `--device` is strictly required if used with `--lb`. It may make sense to instead take a device argument to `--lb` instead to make this very clear.

---

So, if user uses `--lb dev1 --device dev2` we should throw an error, right?

---

Let's throw an error for now as we can't support them together right now but let's also file an issue to make this work. We should be capable of running all  modes:
1. LB only (`--lb eth0`)
2. container networking only (`--device eth0`)
3. LB + container networking on different interfaces (`--lb eth0 --device eth1`)
4. LB + container networking on same interface (`--lb eth0 --device eth0`)

---

Can you check daemon.go:L202-230?